internal/pkg/service/buffer/worker/service/cleanup.go

```diff
@@ -0,0 +1,90 @@
+package service
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/worker/distribution"
+)
+
+type cleanup struct {
+	*Service
+	logger   log.Logger
+	assigner *distribution.Assigner
+
+	lock *sync.RWMutex
+}
+
+func (s *Service) cleanup(ctx context.Context, wg *sync.WaitGroup) <-chan error {
+	return s.dist.StartWork(ctx, wg, s.logger, func(ctx context.Context, assigner *distribution.Assigner) (initDone <-chan error) {
+		return startCleanup(ctx, wg, s, assigner)
+	})
+}
+
+func startCleanup(ctx context.Context, wg *sync.WaitGroup, s *Service, assigner *distribution.Assigner) <-chan error {
+	c := &cleanup{
+		Service:  s,
+		logger:   s.logger.AddPrefix("[cleanup]"),
+		assigner: assigner,
+		lock:     &sync.RWMutex{},
+	}
+
+	// Start ticker
+	startTime := c.clock.Now()
+	c.startTicker(ctx, wg)
+	c.logger.Infof(`initialized | %s`, c.clock.Since(startTime))
+	initDone := make(chan error, 1)
+	defer close(initDone)
+	return initDone
```

Comment:
A druhá věc co si nejsem jistej je to, že tady ta funkce, resp. ani ne tak ona ale ten callback pro `s.dist.StartWork()` o kousek výš, má vracet channel. A tím že tady nemám žádný ty watchery, tak žádný channel nepotřebuju. Vytvořím teda prázdný?